router/internal/batch/batch.go

```diff
@@ -0,0 +1,107 @@
+package batch
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"github.com/goccy/go-json"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"io"
+	"net/http"
+)
+
+const unlimitedBatchEntries = 0
+
+func Handler(maxEntriesPerBatch, maxRoutines int, handlerSent http.Handler, tracerProvider *sdktrace.TracerProvider) http.Handler {
+	tracer := tracerProvider.Tracer(
+		"wundergraph/cosmo/router/internal/batch",
+		trace.WithInstrumentationVersion("0.0.1"),
+	)
+
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		// Read the request body for potential batching.
+		bodyBytes, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, "failed to read request", http.StatusBadRequest)
+			return
+		}
+
+		// Restore the body for further processing (if needed)
+		r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
```

Comment:
Can we avoid parsing the body twice or find a cheaper way to verify if it is a batch request?